nixos/modules/services/libeufin/nexus.nix

```diff
@@ -0,0 +1,172 @@
+
+        This value must be assigned by the bank after having activated a new EBICS subscriber.
+      '';
+      example = "PFC00563";
+    }
+    {
+      name = "IBAN";
+      description = "IBAN of the bank account that is associated with the EBICS subscriber.";
+      example = "CH7789144474425692816";
+    }
+    {
+      name = "BIC";
+      description = "BIC of the bank account that is associated with the EBICS subscriber.";
+      example = "POFICHBEXXX";
+    }
+    {
+      name = "NAME";
+      description = "Legal entity that is associated with the EBICS subscriber.";
+      example = "John Smith S.A.";
+    }
+  ];
+in
+{
+  options.services.libeufin.nexus = {
+    enable = lib.mkEnableOption "EBICS facilitator and component of the libeufin core banking system";
+    package = lib.mkPackageOption pkgs "libeufin" { };
+    debug = lib.mkEnableOption "debug logging";
+
+    settings = lib.mkOption {
+      description = ''
+        Configuration options for the libeufin nexus config file.
+
+        For a list of all possible options, please see the man page [`libeufin-nexus.conf(5)`](https://docs.taler.net/manpages/libeufin-nexus.conf.5.html)
+      '';
+      type = lib.types.submodule {
+        inherit (options.services.libeufin.settings.type.nestedTypes) freeformType;
+        options = {
+          nexus-ebics = lib.pipe mandatoryOptions [
+            (map (option: {
+              inherit (option) name;
+              value = lib.mkOption {
+                type = lib.types.str;
+                default = "You must set `libeufin.nexus.nexus-ebics.${option.name}`";
+                defaultText = "None, you must set this yourself.";
+                inherit (option) description example;
+              };
+            }))
+            lib.listToAttrs
+          ];
+          nexus-httpd = {
+            PORT = lib.mkOption {
+              type = lib.types.port;
+              default = 8084;
+              description = ''
+                The port on which libeufin-bank should listen.
+              '';
+            };
+          };
+          # TODO: should nexus be in a different DB?
+          libeufin-nexusdb-postgres = {
+            CONFIG = lib.mkOption {
+              type = lib.types.str;
+              internal = true;
+              default = "postgresql:///${serviceName}";
+            };
+          };
+        };
+      };
+    };
+  };
+
+  config = lib.mkIf this.enable {
+    services.libeufin = {
+      inherit (this) enable settings;
+    };
+
+    systemd.services =
+      let
+        nexusExe = "${lib.getExe' this.package serviceName}";
+      in
+      {
+        ${serviceName} = {
+          serviceConfig = {
+            DynamicUser = true;
+            User = serviceName;
+            ExecStart = "${nexusExe} serve -c ${configFile}" + lib.optionalString this.debug " -L debug";
+            StateDirectory = serviceName;
+            ReadWritePaths = [ "/var/lib/${serviceName}" ];
```

Comment:
Would it make sense to make this configurable as a `dataDir` option, instead of hard-coding `/var/lib`? See e.g. the [synapse](
https://github.com/NixOS/nixpkgs/tree/717ca7bea5dc1017be222fbba349931b4286786d/nixos/modules/services/matrix/synapse.nix#L1235) moduie. You'd have to ensure it's created though.

Reply:
Yes, this does make sense. Thanks for the suggestion.

Reply:
> You'd have to ensure it's created though.

I think this is already handled by systemd.